static/js/components/SourceSearchBar.jsx

```diff
@@ -0,0 +1,176 @@
+    color: theme.palette.info.main,
+    "& > * + *": {
+      marginLeft: theme.spacing(2),
+    },
+  },
+}));
+
+function useDebouncer(value, delay) {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+  return debouncedValue;
+}
+
+const SourceSearchBar = () => {
+  const dispatch = useDispatch();
+  const [options, setOptions] = useState([]);
+  const [value, setValue] = useState(null);
+  const [inputValue, setInputValue] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [open, setOpen] = useState(false);
+  const navigate = useNavigate();
+
+  const classes = useStyles();
+
+  const debouncedInputValue = useDebouncer(inputValue, 500);
+  const cache = useRef({});
+
+  useEffect(() => {
+    const get = (val) =>
+      dispatch(
+        GET(
+          `/api/sources?sourceID=${val}&pageNumber=1&totalMatches=25&includeComments=false&removeNested=true`,
+          "skyportal/FETCH_AUTOCOMPLETE_SOURCES"
+        )
+      );
+    (async () => {
+      if (debouncedInputValue === "") {
+        setOptions([]);
+        setOpen(false);
+        return undefined;
+      }
+
+      let newOptions = [];
+
+      if (cache.current[debouncedInputValue]) {
+        setLoading(true);
+        newOptions = cache.current[debouncedInputValue];
+        setLoading(false);
+      } else {
+        setLoading(true);
+        const response = await get(debouncedInputValue);
+        setLoading(false);
+
+        const matchingSources = await response.data.sources;
+
+        if (matchingSources) {
+          const rez = Object.keys(matchingSources).map(
+            (key) => matchingSources[key].id
+          );
+          newOptions = [...newOptions, ...rez];
+        }
+        cache.current[debouncedInputValue] = newOptions;
+      }
+      setOpen(true);
+      setOptions(newOptions);
+      return undefined;
+    })();
+  }, [dispatch, debouncedInputValue]);
+
+  return (
+    <Autocomplete
+      color="primary"
+      id="source-search-bar"
+      style={{ padding: "0.3rem" }}
+      classes={{ root: classes.root, paper: classes.paper }}
+      isOptionEqualToValue={(option, val) => option.name === val.name}
+      getOptionLabel={(option) => option}
+      onInputChange={(e, val) => setInputValue(val)}
+      onChange={(event, newValue, reason) => {
+        setValue(newValue);
+        if (reason === "selectOption") {
+          navigate(`/source/${newValue}`);
+        }
+        if (reason === "clear") {
+          setOpen(false);
+        }
+      }}
+      onClose={() => setOpen(false)}
+      size="small"
+      noOptionsText="No matching sources."
+      options={options}
+      open={open}
+      loading={loading}
+      clearOnEscape
+      clearOnBlur
```

Comment:
I'm not sure if we want the search bar to automatically clear when clicking outside of it (I assume that's what `clearOnBlur` does.

Reply:
You're right, I should delete this option !